src/NServiceBus.Metrics/Probes/SignalProbeBuilder.cs

```diff
@@ -0,0 +1,27 @@
+using System.Reflection;
+using NServiceBus;
+using NServiceBus.Features;
+using NServiceBus.Metrics;
+
+
+
+abstract class SignalProbeBuilder
+{
+    protected abstract void WireUp(FeatureConfigurationContext context, SignalProbe probe);
+
+    public SignalProbe Build(FeatureConfigurationContext context)
+    {
+        var probe = GetProbe();
+
+        WireUp(context, probe);
+
+        return probe;
+    }
+
+    SignalProbe GetProbe()
+    {
+        var attr = GetType().GetCustomAttribute<ProbePropertiesAttribute>();
```

Comment:
This requires a null check, and throwing a user friendly exception like below, if not attribute found.

$The type '{GetType()}' is not annotated with 'ProbePropertiesAttribute'. Add this attribute for providing probe's metadata